### DISCUSSION TITLE
How to obtain the number and type of parameters in the ABI of a non open source contract

### DISCUSSION THREAD

**Discussion by nick199910** [cypherpunk-symposium/dark-forest-toolkit#33] [Q&A]
How to obtain the number and type of parameters in the ABI of a non open source contract

**Comment by von-steinkirch** [comment]
good question.
let's see whether the channel engage, otherwise i will give you my own take in a couple of weeks.

**Comment by monperrus** [comment]
see paper "SigRec: Automatic Recovery of Function Signatures in Smart Contracts" 
https://www4.comp.polyu.edu.hk/~csxluo/SigRec.pdf
